library/src/main/java/com/novoda/downloadmanager/DownloadBatchRequirementRule.java

```diff
@@ -0,0 +1,18 @@
+package com.novoda.downloadmanager;
+
+/**
+ * Used to define the rules that need to be satisfied to download an batch.
+ * Clients of this library can create their own custom implementations and
+ * pass them to {@link DownloadManagerBuilder#withDownloadBatchRequirementRules(DownloadBatchRequirementRule...)}.
+ */
+public interface DownloadBatchRequirementRule {
+
+    /**
+     * Determines whether a {@link DownloadBatch} has violated a size requirement rule.
+     *
+     * @param batchSize of the batch to check.
+     * @param downloadBatchStatus of batch to check.
+     * @return whether the {@link DownloadBatch} has violated the rule.
+     */
+    boolean hasViolatedRule(DownloadBatchStatus downloadBatchStatus, Long batchSize);
```

Comment:
You should only need `DownloadBatchStatus` as it already contains the batch size.

Reply:
Good point. I haven't noticed that before